crates/oxc_minifier/src/ast_passes/replace_global_defines.rs

```diff
@@ -19,7 +19,12 @@ pub struct ReplaceGlobalDefinesConfig(Arc<ReplaceGlobalDefinesConfigImpl>);
 #[derive(Debug)]
 struct ReplaceGlobalDefinesConfigImpl {
     identifier_defines: Vec<(/* key */ String, /* value */ String)>,
-    // TODO: dot defines
+    dot_defines: Vec<(/* member expression parts */ Vec<String>, /* value */ String)>,
```

Comment:
these can probably be optimized into hashmaps.

changing them into hash maps will also allow us to detect collisions easier

e.g. `ReplaceGlobalDefinesConfig::new(&[("foo", "bar"), ("foo","baz")])`

should probably report an error since `foo=baz` will never be respected